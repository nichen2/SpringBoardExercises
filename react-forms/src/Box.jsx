const Box = ({id, width, height, backgroundColor, removeBox}) => {

    const style = {
        width: width,
        height: height,
        backgroundColor: backgroundColor
    }
    return (    
        <div style={style}>
            <button onClick={() => removeBox(id)}>X</button>
        </div>
    )
}

Box.defaultProps = {
    width: "100px",
    height: "100px",
    backgroundColor: "cyan"
}

export default Box