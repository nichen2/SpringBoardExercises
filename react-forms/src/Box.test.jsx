import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import Box from './Box';

describe('Box Component', () => {
  it('renders without crashing', () => {
    render(<Box width="100px" height="100px" backgroundColor="red" />);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Box width="100px" height="100px" backgroundColor="red" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
