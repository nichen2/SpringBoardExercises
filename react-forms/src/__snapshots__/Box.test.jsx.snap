// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Box Component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="width: 100px; height: 100px; background-color: red;"
  >
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
